### PR TITLE
fixed most of security warnings

### PR DESCRIPTION
**Pull request by khoinguyen-hub**
@Andrey1994  I've fixed most of the warnings. The ones that have to do with exception; however, I don't really know how to handle them so I put them as `except BaseException:`. I would love to hear any feedback and any tips or suggestions to change the code before the merge.

### CODE REVIEW

**Comment by Andrey1994** [comment]
Thanks, I will review it tomorrow, in the meantime you can fix failed tests. For clang-format take a look at https://brainflow.readthedocs.io/en/stable/BrainFlowDev.html#code-style

**Comment by Andrey1994** [comment]
Also, CI compiles it with warnings as errors option, to reproduce it locally you can add `--warnings-as-errors` argument to build.py

**Comment by khoinguyen-hub** [comment]
I'm having trouble with Cmake and building it in general. I think because I don't have Visual Studio 2019 installed which I am getting it now.

File: src/board_controller/inc/brainflow_input_params.h [hunk 1]
```diff
@@ -30,25 +30,25 @@ struct BrainFlowInputParams
     }
 
     // default copy constructor and assignment operator are ok, need less operator to use in map
-    bool operator< (const struct BrainFlowInputParams &other) const
```
Comment (Andrey1994) [inline src/board_controller/inc/brainflow_input_params.h:33]: why is it changed?

File: src/ml/inc/brainflow_model_params.h [hunk 2]
```diff
@@ -21,7 +21,7 @@ struct BrainFlowModelParams
     }
 
     // default copy constructor and assignment operator are ok, need less operator to use in map
-    bool operator< (const struct BrainFlowModelParams &other) const
```
Comment (Andrey1994) [inline src/ml/inc/brainflow_model_params.h:24]: here too

**Review by Andrey1994 — COMMENTED** [review]
Looks good so far, let's fix the remaining recommendations

**Comment by Andrey1994** [comment]
Exception handling is fine

File: src/ml/inc/brainflow_model_params.h (hunk 2 above)
Comment (khoinguyen-hub) [reply src/ml/inc/brainflow_model_params.h:24]: @Andrey1994 i think i changed these on accdient while changing the other functions arguments.

Comment (khoinguyen-hub) [reply src/ml/inc/brainflow_model_params.h:24]: They should still be const and pass by value.

Comment (Andrey1994) [reply src/ml/inc/brainflow_model_params.h:24]: Could you restore it here and in other places?

Comment (khoinguyen-hub) [reply src/ml/inc/brainflow_model_params.h:24]: ok i will change them back

**Review by Andrey1994 — APPROVED** [review]

**Comment by Andrey1994** [comment]
Thanks for fixing it, there are a few more recommendations from lgtm you are welcome to fix them in another PR

**Pull request merged**